File: src/main/java/spring/webmvc/infrastructure/config/AsyncConfig.java
package spring.webmvc.infrastructure.config;

import java.util.concurrent.Executor;

import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.scheduling.annotation.EnableAsync;
import org.springframework.scheduling.concurrent.ThreadPoolTaskExecutor;

@EnableAsync
@Configuration(proxyBeanMethods = false)
public class AsyncConfig {

	@Bean
	public Executor threadPoolTaskExecutor() {
		ThreadPoolTaskExecutor taskExecutor = new ThreadPoolTaskExecutor();

		taskExecutor.setCorePoolSize(3);
		taskExecutor.setMaxPoolSize(10);
		taskExecutor.setQueueCapacity(100);
		taskExecutor.setThreadNamePrefix("Executor-");

		return taskExecutor;
	}
}

File: spring-web-kotlin/src/main/kotlin/spring/web/kotlin/infrastructure/config/AsyncConfig.kt
package spring.web.kotlin.infrastructure.config

import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.scheduling.annotation.EnableAsync
import org.springframework.scheduling.concurrent.ThreadPoolTaskExecutor

@EnableAsync
@Configuration(proxyBeanMethods = false)
class AsyncConfig {

    @Bean
    fun threadPoolTaskExecutor() = ThreadPoolTaskExecutor()
        .apply {
            corePoolSize = 3
            maxPoolSize = 10
            queueCapacity = 100
            setThreadNamePrefix("Executor-")
        }
}